### DISCUSSION TITLE
Notification from background service

### DISCUSSION THREAD

**Discussion by avojacek** [microsoft/fluentui-blazor#1840] [Q&A] [answered]
Hi all,
how to correctly display notifications from background task as a Toast message or MessageBar?
I have some background tasks which are queued in background service, they are started from Blazor app. How can I display message to user using MessageBar or Toast? Is this even possible to do that. Is there way to display those only to user which started that job or even better to all who is using Blazor app at moment of completion that task?
My blazor App is server side no wasm.

Thank you a lot.
A.

**Comment by dvoituron** [comment, marked as answer]
You could use the [Toast](https://www.fluentui-blazor.net/Toast) service or the [Message](https://www.fluentui-blazor.net/MessageBar) service. Every time you will add a message, Blazor will send a SignalR message to the client to display your information.

If your process come from the "startup tasks", you should use the [SplashScreen](https://www.fluentui-blazor.net/SplashScreen) component.

**Reply by avojacek** [reply]
Great; but is it ok, to Inject toast/message service into backgroud task?

**Reply by dvoituron** [reply]
I don't quite understand your question. These services are pieces of C# code that you can inject wherever you like.

If your background task is running in a microservice, on another server, obviously SignalR won't be able to send a message to the client. But if your background task is a `Task.Run` in your project, there shouldn't be any problem.

It's more a question of C#/Blazor than FluentUI Lib, I think.

**Reply by avojacek** [reply]
I'm sorry. I did not explain that well (sorry for my english) .
What I ment by bacground task was HostedService (in same process as Blazor app) inherited from BacgroudService which process queued work. 
So that question had to be,  Is it safe inject say MessageService into BackgroundService.

**Reply by dvoituron** [reply]
English isn't my main language either.

Yes, you can inject these services (which is already the case via `builder.Services.AddFluentUIComponents()`) 
and use them in your application.

**Reply by vnbaaij** [reply]
No need to apologize. 

As Denis said. it is more a C#/Blazor thing. If the background service has 'access to' or can 'see' the service, then yes it can be injected. 
But I don't think it will work because the background service probably runs on a different thread than the site/app itself. Not sure if it will be picked up. But you can just try it to find out, right.

**Reply by avojacek** [reply]
Yes, I will try, I tried to look into AddFluentUIComponents and MessageService is registered as scoped, so I thing it will not work. I have to find another solution.
Anyway thank you both.

**Reply by dvoituron** [reply]
In this case, don't call `AddFluentUIComponents` and check this method to create a similar one with your preferences. This is just an Helper method to inject these services.

**Reply by avojacek** [reply]
That's looks promising :-) I was scared to do that. I will look at that if I can register that as Singleton. Thank you a lot.

**Reply by avojacek** [reply]
Hm :-) no luck MessageService needs NavigationManager which is scoped and cannot be consumed by singleton. :-) But anyway thank you a lot. I will do that some other way.
It is realy usefull to look into source code :-) 
Thank you both for your time.
Ales
